From 56eba126ee1286a23799976c2d4e6463bd58d6c7 Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Fri, 29 Aug 2025 23:58:54 -0400
Subject: [PATCH 6/6] TST: Use a temporary directory for
 test_save_figure_return

This avoids having to manually clean up the resulting file, which seems
flaky.

Signed-off-by: Elliott Sales de Andrade <quantum.analyst@gmail.com>
---
 lib/matplotlib/tests/test_backend_qt.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/matplotlib/tests/test_backend_qt.py b/lib/matplotlib/tests/test_backend_qt.py
index 962f5ce942..e191dd73eb 100644
--- a/lib/matplotlib/tests/test_backend_qt.py
+++ b/lib/matplotlib/tests/test_backend_qt.py
@@ -219,14 +219,15 @@ def test_figureoptions():
 
 
 @pytest.mark.backend('QtAgg', skip_on_importerror=True)
-def test_save_figure_return():
+def test_save_figure_return(tmp_path):
+    os.chdir(tmp_path)
     fig, ax = plt.subplots()
     ax.imshow([[1]])
     prop = "matplotlib.backends.qt_compat.QtWidgets.QFileDialog.getSaveFileName"
     with mock.patch(prop, return_value=("foobar.png", None)):
         fname = fig.canvas.manager.toolbar.save_figure()
-        os.remove("foobar.png")
         assert fname == "foobar.png"
+        assert (tmp_path / "foobar.png").exists()
     with mock.patch(prop, return_value=(None, None)):
         fname = fig.canvas.manager.toolbar.save_figure()
         assert fname is None
-- 
2.50.0

